text/0000-rvalue_static_promotion.md

```diff
@@ -0,0 +1,171 @@
+- Feature Name: rvalue_static_promotion
+- Start Date: 2015-12-18
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+Promote constexpr rvalues to values in static memory instead of
+stack slots, and expose those in the language by being able to directly create
+`'static` references to them. This would allow code like
+`let x: &'static u32 = &42` to work.
+
+# Motivation
+[motivation]: #motivation
+
+Right now, when dealing with constant values, you have to explicitly define
+`const` or `static` items to create references with `'static` lifetime,
```

Comment:
For example, `&None::<T>` could be returned even if `T` were a type parameter, or `&[T; 0]` (but the array case is actually hardcoded from before any attempts at rvalue promotions).